### PR TITLE
Fix size_of_val and support min_align_of_val

### PR DESCRIPTION
**Pull request by HKalbasi**

### CODE REVIEW

File: crates/hir-ty/src/mir/eval/shim.rs [hunk 1]
```diff
@@ -186,44 +190,24 @@ impl Evaluator<'_> {
...
-                            self.read_memory(piece_ptr_addr.offset(ptr_size), ptr_size)
-                                .ok()?
-                                .try_into()
-                                .ok()?,
-                        );
-                        let piece_data = self.read_memory(piece_addr, piece_len).ok()?;
-                        message += &std::string::String::from_utf8_lossy(piece_data);
-                    }
-                    Some(message)
+                    let x = self.db.crate_def_map(self.crate_id).crate_root();
```
Comment (Veykril) [inline crates/hir-ty/src/mir/eval/shim.rs:193]: `x` isn't really a great variable name, especially in this case here. You tend to do that a lot which is why I'm raising that here. (and in cases where we need a quick non describing name like in mapping closures etc we tend to use `it`, https://github.com/rust-lang/rust-analyzer/blob/master/docs/dev/style.md#variable-naming)

Comment (HKalbasi) [reply crates/hir-ty/src/mir/eval/shim.rs:193]: I will make a de-x-ify follow up PR to remove all xs everywhere.

**Pull request merged**